core/domain/learner_progress_services.py

```diff
@@ -289,6 +354,64 @@ def mark_exploration_as_incomplete(
         _save_incomplete_activities(incomplete_activities)
 
 
+def mark_story_as_incomplete(user_id, story_id):
+    """Adds the story id to the incomplete list of the user unless the
+    story has been already completed by the user. If the story is already
+    present in the incomplete list, just the details associated with it are
+    updated.
```

Comment:
Nothing is updated if the story is already in the incomplete list, rigth? Update the docstring.

Reply:
Done.